src/EnergyPlus/IndoorGreen.cc

```diff
@@ -0,0 +1,778 @@
+                                   format("EMS/Python Plugin for ET Data Driven Model not find in {}={}",
+                                          cCurrentModuleObject,
+                                          lw->indoorgreen(IndoorGreenNum).IndoorGreenName));
+                   }
+                } 
+                Real64 effectivearea = std::min(lw->indoorgreen(IndoorGreenNum).LeafArea,
+                                                LAI*state.dataSurface->Surface(lw->indoorgreen(IndoorGreenNum).SurfPtr).Area); 
+                ETTotal = lw->indoorgreen(IndoorGreenNum).ETRate * Timestep *effectivearea*
+                           ScheduleManager::GetCurrentScheduleValue(state, lw->indoorgreen(IndoorGreenNum).SchedPtr); //kg; this unit area should be surface area instead of total leaf area
+                Real64 hfg = Psychrometrics::PsyHfgAirFnWTdb(ZonePreHum, ZonePreTemp) / std::pow(10, 6); // Latent heat of vaporization (MJ/kg)
+                lw->indoorgreen(IndoorGreenNum).LambdaET =
+                    ETTotal * hfg * std::pow(10, 6) / state.dataSurface->Surface(lw->indoorgreen(IndoorGreenNum).SurfPtr).Area /
+                    Timestep; // (W/m2))
+                rhoair = Psychrometrics::PsyRhoAirFnPbTdbW(state, state.dataEnvrn->OutBaroPress, ZonePreTemp, ZonePreHum);
+                ZoneAirVol = state.dataHeatBal->Zone(lw->indoorgreen(IndoorGreenNum).ZonePtr).Volume;
+                ZoneNewHum =
+                ZonePreHum + ETTotal / (rhoair * ZoneAirVol);
+                Twb = Psychrometrics::PsyTwbFnTdbWPb(state, ZonePreTemp, ZonePreHum, state.dataEnvrn->OutBaroPress);
+                ZoneSatHum = Psychrometrics::PsyWFnTdpPb(state, ZonePreTemp, state.dataEnvrn->OutBaroPress); //saturated humidity ratio
+                HCons = Psychrometrics::PsyHFnTdbW(ZonePreTemp, ZonePreHum);
+                if (ZoneNewHum <= ZoneSatHum) {
+                ZoneNewTemp = Psychrometrics::PsyTdbFnHW(HCons, ZoneNewHum);
+                } else {
+                ZoneNewTemp = Twb;
+                ZoneNewHum = ZoneSatHum;
+                }
+                HMid = Psychrometrics::PsyHFnTdbW(ZoneNewTemp, ZoneNewHum);
+                lw->indoorgreen(IndoorGreenNum).SensibleRate =
+                    (1-lw->indoorgreen(IndoorGreenNum).LEDRadFraction) * lw->indoorgreen(IndoorGreenNum).LEDActualEleP; // convective heat gain from LED lights when LED is on; heat convection from plants was considered and counted from plant surface heat balance.
+                lw->indoorgreen(IndoorGreenNum).LatentRate = ZoneAirVol * rhoair * (HCons - HMid) / Timestep;   // unit W
+                state.dataHeatBalSurf->SurfQAdditionalHeatSourceInside(lw->indoorgreen(IndoorGreenNum).SurfPtr) =
+                    -1.0*lw->indoorgreen(IndoorGreenNum).LambdaET;
+            }
+    }
+
+
+    Real64 ETPenmanMonteith(EnergyPlusData &state, Real64 &ZonePreTemp, Real64 &ZonePreHum, Real64 &ZoneCO2, Real64 &ZonePPFD, Real64 &ZoneVPD, Real64 &LAI)
```

Comment:
Except for `state`, these are scalar inputs. Pass them by value, not reference. Reference is for output parameters.

Reply:
Thank you! Removed the reference & in Penman Monteith ET model inputs